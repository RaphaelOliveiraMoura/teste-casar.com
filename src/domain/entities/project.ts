export type ProjectDto = {
  id: string;
  title: string;
  description: string;
  updatedAt: string;
  techs: string[];
  favorite?: boolean;
};

export class Project {
  public id!: string;
  public title!: string;
  public description!: string;
  public updatedAt!: Date;
  public techs!: string[];
  public favorite?: boolean;

  constructor(props: ProjectDto) {
    Object.assign(this, props);
    this.updatedAt = new Date(props.updatedAt);
  }

  setFavorite(isFavorite: boolean) {
    this.favorite = isFavorite;
  }

  toObject() {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      updatedAt: this.updatedAt.toISOString(),
      techs: this.techs,
      favorite: this.favorite,
    };
  }
}
