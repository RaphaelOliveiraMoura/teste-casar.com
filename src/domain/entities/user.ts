export interface User {
  identifier: string;
  name: string;
  description: string;
  imageUrl: string;
}
