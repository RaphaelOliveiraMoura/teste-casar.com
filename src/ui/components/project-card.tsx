import { Heart } from "lucide-react";

import { Project } from "@/domain/entities/project";

type ProjectCardProps = {
  project: Project;
};

export function ProjectCard({ project }: ProjectCardProps) {
  return (
    <article className="flex gap-4 rounded-sm border p-4">
      <div className="grid flex-1 gap-2">
        <h2 className="text-lg font-semibold">{project.title}</h2>
        <p className="max-w-[700px]">{project.description}</p>

        <div className="flex gap-4 text-sm">
          <div className="flex items-center gap-1">
            <div className="size-4 rounded-full bg-blue-500"></div>
            <span>Typescript</span>
          </div>

          <div>
            Updated on{" "}
            {project.updatedAt.toLocaleDateString(undefined, {
              dateStyle: "medium",
            })}
          </div>
        </div>
      </div>
      <div>
        <div className="rounded-full border border-primary p-3">
          <Heart className="fill-primary text-primary" />
        </div>
      </div>
    </article>
  );
}
