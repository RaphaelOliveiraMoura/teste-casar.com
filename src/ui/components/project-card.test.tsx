import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";

import { Project } from "@/domain/entities/project";

import { ProjectCard } from "./project-card";

test("<ProjectCard />", () => {
  const snapshot = render(
    <ProjectCard
      project={
        new Project({
          id: "1",
          title: "mock project",
          description: "project description",
          techs: ["JavaScript"],
          updatedAt: new Date().toISOString(),
          favorite: false,
        })
      }
    />,
  );

  expect(screen.getByText("mock project")).toBeDefined();
  expect(screen.getByText("project description")).toBeDefined();
  expect(screen.getByText("JavaScript")).toBeDefined();

  expect(snapshot).toMatchSnapshot();
});
