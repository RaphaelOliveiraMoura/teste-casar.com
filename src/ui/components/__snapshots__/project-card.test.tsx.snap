// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProjectCard /> 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="flex gap-4 rounded-sm border p-4"
      >
        <div
          class="grid flex-1 gap-2"
        >
          <h2
            class="text-lg font-semibold"
          >
            mock project
          </h2>
          <p
            class="max-w-[700px]"
          >
            project description
          </p>
          <div
            class="flex flex-col gap-4 text-sm md:flex-row"
          >
            <div
              class="flex items-center gap-1"
            >
              <div
                class="size-4 rounded-full bg-[#F5DA79]"
              />
              <span>
                JavaScript
              </span>
            </div>
            <div>
              Updated on
               
              Jun 16, 2024
            </div>
          </div>
        </div>
        <div>
          <button
            aria-label="Add as favorite"
            class="rounded-full border p-3 bg-white-background-matte"
            type="button"
          >
            <svg
              class="lucide lucide-heart text-placeholder"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
              />
            </svg>
          </button>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="flex gap-4 rounded-sm border p-4"
    >
      <div
        class="grid flex-1 gap-2"
      >
        <h2
          class="text-lg font-semibold"
        >
          mock project
        </h2>
        <p
          class="max-w-[700px]"
        >
          project description
        </p>
        <div
          class="flex flex-col gap-4 text-sm md:flex-row"
        >
          <div
            class="flex items-center gap-1"
          >
            <div
              class="size-4 rounded-full bg-[#F5DA79]"
            />
            <span>
              JavaScript
            </span>
          </div>
          <div>
            Updated on
             
            Jun 16, 2024
          </div>
        </div>
      </div>
      <div>
        <button
          aria-label="Add as favorite"
          class="rounded-full border p-3 bg-white-background-matte"
          type="button"
        >
          <svg
            class="lucide lucide-heart text-placeholder"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
            />
          </svg>
        </button>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
